### PR TITLE
feat: apply juju secrets to transfer bind account password

### PR DESCRIPTION
**Pull request by wood-push-melon**
This pull request tries to use the Juju secret to encapsulate the bind account password passed to the requirer charm. The main logic is generally done in the `ldap` charm library so that the provider charm does not need to know it.

Please note that I intentionally leave the requirer side this time. Will chat with the stakeholders about how they wanna deal with the secret.

### CODE REVIEW

File: src/integrations.py
Comment (wood-push-melon) [inline src/integrations.py:154]: For internal k8s-cluster use.

**Review by BarcoMasile — APPROVED** [review]
LGTM, I'd wait for another review given my limited knowledge on this though!

File: lib/charms/glauth_k8s/v0/ldap.py
Comment (natalian98) [inline lib/charms/glauth_k8s/v0/ldap.py:129]: I think you only need `from ops import Secret`

**Review by natalian98 — APPROVED** [review]
one minor comment, other than that lgtm :+1:

File: lib/charms/glauth_k8s/v0/ldap.py
Comment (wood-push-melon) [reply lib/charms/glauth_k8s/v0/ldap.py:129]: oh, I can not do that because the naming will conflict with the class `Secret`.

**Pull request merged**
